pkg/operator/observe_config.go

```diff
@@ -107,14 +129,16 @@ func (c ConfigObserver) sync() error {
 
 // observeEtcdEndpoints reads the etcd endpoints from the endpoints object and then manually pull out the hostnames to
 // get the etcd urls for our config. Setting them observed config causes the normal reconciliation loop to run
-func observeEtcdEndpoints(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+// REQUIRED EndPoints etcd.kube-system
+// REQUIRED etcd.kube-system/subsets[], len > 0
+func observeEtcdEndpoints(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, []error) {
 	etcdURLs := []string{}
 	etcdEndpoints, err := kubeClient.CoreV1().Endpoints(etcdNamespaceName).Get("etcd", metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		return observedConfig, nil
+		return observedConfig, []error{fmt.Errorf("Endpoints/etcd.kube-system not found")}
```

Comment:
This doesn't appear to match your proposal.  Aren't you supposed to be returning the map with the old value set?

Reply:
Not for the errors.